homeassistant/components/insteon/api/scenes.py

```diff
@@ -0,0 +1,122 @@
+"""Web socket API for Insteon scenes."""
+
+from pyinsteon import devices
+from pyinsteon.constants import ResponseStatus
+from pyinsteon.managers.scene_manager import (
+    DeviceLinkSchema,
+    async_add_or_update_scene,
+    async_delete_scene,
+    async_get_scene,
+    async_get_scenes,
+)
+import voluptuous as vol
+
+from homeassistant.components import websocket_api
+from homeassistant.core import HomeAssistant
+
+from ..const import ID, TYPE
+
+
+def _scene_to_dict(scene):
+    """Return a dictionary mapping of a scene."""
+    device_dict = {}
+    for addr, links in scene["devices"].items():
+        str_addr = str(addr)
+        device_dict[str_addr] = []
+        for data in links:
+            device_dict[str_addr].append(
+                {
+                    "data1": data.data1,
+                    "data2": data.data2,
+                    "data3": data.data3,
+                    "has_controller": data.has_controller,
+                    "has_responder": data.has_responder,
+                }
+            )
+    return {"name": scene["name"], "group": scene["group"], "devices": device_dict}
+
+
+@websocket_api.websocket_command({vol.Required(TYPE): "insteon/scenes/get"})
+@websocket_api.require_admin
```

Comment:
Should admin be required also to get scenes?

Reply:
Yes. This is an intermediate release to enable scene creating / editing, which is an admin function. There is another change coming to make it easier for a standard user to consume a scene after it is created.